### PR TITLE
check raise invalid shape of _log_weight_as_image

### PR DESCRIPTION
**Pull request by autoih**
Raise ValueError in _log_weight_as_image()

### CODE REVIEW

File: tensorflow/python/keras/callbacks.py [hunk 1]
```diff
@@ -1723,6 +1723,8 @@ def _log_weight_as_image(self, weight, weight_name, epoch):
         w_img = array_ops.transpose(w_img, perm=[2, 0, 1])
         shape = K.int_shape(w_img)
       w_img = array_ops.reshape(w_img, [shape[0], shape[1], shape[2], 1])
+    elif len(shape) != 4:
+      raise ValueError('Invalid shape: {}'.format(repr(shape)))
```
Comment (omalleyt12) [inline tensorflow/python/keras/callbacks.py]: It's actually common for people to want to visualize whatever weights of their Model can be visualized, while still having weights that can't be visualized. I think silently skipping weights that can't be visualized is the right approach here

**Review by omalleyt12 — COMMENTED** [review]
Thanks for the PR! I think in this case we actually have to leave as-is

File: tensorflow/python/keras/callbacks.py (hunk 1 above)
Comment (I-Hong) [reply tensorflow/python/keras/callbacks.py]: So, can we update the document then?

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**